core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1463,6 +1470,223 @@ class KafkaApis(val requestChannel: RequestChannel,
     sendResponseMaybeThrottle(request, createResponse)
   }
 
+  def handleListAcls(request: RequestChannel.Request): Unit = {
+    authorizeClusterAction(request)
+    val listAclsRequest = request.body[ListAclsRequest]
+    authorizer match {
+      case None => {
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new ListAclsResponse(throttleTimeMs, new SecurityDisabledException(
+            "No Authorizer is configured on the broker."), Collections.emptySet[AclFixture]);
+        sendResponseMaybeThrottle(request, createResponse)
+      }
+      case Some(auth) => {
+        val filter = listAclsRequest.filter()
+        var returnedAcls = new util.ArrayList[AclFixture]
+        val aclMap : Map[Resource, Set[Acl]] = auth.getAcls()
+        aclMap.foreach {
+          case (resource, acls) => {
+            acls.foreach {
+              case (acl) => {
+                val fixture = new AclFixture(new AclResource(resource.resourceType.toJava, resource.name),
+                               new AclData(acl.principal.toString(), acl.host.toString(),
+                                           acl.operation.toJava, acl.permissionType.toJava))
+                if (filter.matches(fixture))
+                  returnedAcls.add(fixture)
+              }
+            }
+          }
+        }
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new ListAclsResponse(throttleTimeMs, null, returnedAcls)
+        sendResponseMaybeThrottle(request, createResponse)
+      }
+    }
+  }
+
+  /**
+    * Convert an ACL fixture to a Scala object.
+    * All ACL and resource fields must be specified (no UNKNOWN, ANY, or null fields are allowed.)
+    *
+    * @param fixture    The fixture as a Java object.
+    * @return           The fixture as a scala object, or an exception if there was an error
+    *                   converting the Java object.
+    */
+  def fixtureToScala(fixture: AclFixture) : Try[(Resource, Acl)] = {
+    fixture.resource().resourceType() match {
+      case AclResourceType.ANY => return Failure(new InvalidRequestException("Invalid resource type"))
+      case AclResourceType.UNKNOWN => return Failure(new InvalidRequestException("Invalid resource type"))
+      case _ => ()
+    }
+    var resourceType: ResourceType = null
+    try {
+      resourceType = ResourceType.fromJava(fixture.resource().resourceType)
+    } catch {
+      case throwable: Throwable => return Failure(new InvalidRequestException("Unknown resource type"))
```

Comment:
OK.  I can convert them to return Try[...].  The exception text will include the value and what the possible valid values are.